### ISSUE TITLE
Generate URLs for routes

### ISSUE THREAD

**Issue by slavafomin** [balderdashy/sails#3151]
Hello!

It looks like Sails doesn't provide a way to generate URLs for routes. It means, that developer needs to provide URLs in several places of the application (in route configuration, in links, in scripts, etc). This dramatically breaks the DRY principle. I do believe that URL generation is a very important aspect of a router component for any solid web framework. For example [Symfony 2](http://symfony.com/doc/current/book/routing.html#generating-urls) and [Angular.js](http://angular-ui.github.io/ui-router/site/#/api/ui.router.state.directive:ui-sref) knows how to do this.

In order to implement this we will need a way to specify unique names for routes and URL generator should be both available as a service and as a view helper.

What do you think?

**Comment by tjwebb** [comment]
> It looks like Sails doesn't provide a way to generate URLs for routes.

[It sure does!](http://sailsjs.org/features#?rest). Routes are automatically generated from your ORM. Make sure you take a look at the Blueprint API docs: http://sailsjs.org/documentation/reference/blueprint-api

**Comment by slavafomin** [comment]
Hello @tjwebb, I think you misunderstood my issue. I were taking about a way to generate URLs to the specified routes dynamically to use them in the views. Please see this chapter for example: http://symfony.com/doc/current/book/routing.html#generating-urls-from-a-template.

**Comment by sailsbot** [comment]
Thanks for posting, @slavafomin.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by RomainLanz** [comment]
Any news?
